subsys/net/l2/ethernet/lldp/lldp.c

```diff
@@ -111,6 +111,15 @@ static int lldp_send(struct ethernet_lldp *lldp)
 		goto out;
 	}
 
+	if (lldp->optional_du && lldp->optional_len) {
```

Comment:
you will need modify line 100 the size requested for allocation : basically sizeof(struct net_llpdu) + lldp->optional_len